<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="tkik" />


<title>Differential methlytion AML</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">22_AEL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Differential methlytion AML</h1>
<h4 class="author">tkik</h4>
<h4 class="date">16 June, 2023</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-06-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>22_AEL/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0.4). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230201code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230201)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230201code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230201)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtkikAELmethylationtree83a5ebf76ef26bb0e11b9b21d4606530b6df186etargetblank83a5ebfa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tkik/AEL_methylation/tree/83a5ebf76ef26bb0e11b9b21d4606530b6df186e" target="_blank">83a5ebf</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtkikAELmethylationtree83a5ebf76ef26bb0e11b9b21d4606530b6df186etargetblank83a5ebfa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tkik/AEL_methylation/tree/83a5ebf76ef26bb0e11b9b21d4606530b6df186e" target="_blank">83a5ebf</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    refdir/

Untracked files:
    Untracked:  analysis/3_combine_meth_ge_mofa.Rmd
    Untracked:  analysis/DMR_report_illumina.Rmd
    Untracked:  analysis/candidate_genes.Rmd
    Untracked:  analysis/enhancer_gene_pairs.Rmd
    Untracked:  analysis/epiCMIT_score.Rmd
    Untracked:  analysis/homer_all.Rmd
    Untracked:  analysis/homer_results_promoter.Rmd
    Untracked:  analysis/inital_analysis.Rmd
    Untracked:  analysis/prepare_homer.Rmd
    Untracked:  analysis/test.Rmd
    Untracked:  code/dataETOC2.Rd
    Untracked:  code/support_functions.R
    Untracked:  data/AEL_gr1_vs._AEL_gr2_down.txt
    Untracked:  data/AEL_gr1_vs._AEL_gr2_up.txt
    Untracked:  data/AEL_subgroup_dmr.bedGraph
    Untracked:  data/CMP_vs._all_in_normal_down.txt
    Untracked:  data/CMP_vs._all_in_normal_up.txt
    Untracked:  data/CMP_vs._all_in_tumor_down.txt
    Untracked:  data/CMP_vs._all_in_tumor_up.txt
    Untracked:  data/DML_tests.RDS
    Untracked:  data/GSE63409_sample_sheet.txt
    Untracked:  data/HSC_vs._all_in_normal_down.txt
    Untracked:  data/HSC_vs._all_in_normal_up.txt
    Untracked:  data/HSC_vs._all_in_tumor_down.txt
    Untracked:  data/HSC_vs._all_in_tumor_up.txt
    Untracked:  data/MPP_vs._all_in_normal_down.txt
    Untracked:  data/MPP_vs._all_in_normal_up.txt
    Untracked:  data/MPP_vs._all_in_tumor_down.txt
    Untracked:  data/MPP_vs._all_in_tumor_up.txt
    Untracked:  data/bigWigs_2023/
    Untracked:  data/deseq.RDS
    Untracked:  data/dmls_single.RDS
    Untracked:  data/dmls_single_15_06.RDS
    Untracked:  data/dmr_beds/
    Untracked:  data/dmr_beds_enhancer/
    Untracked:  data/dmr_beds_promoter/
    Untracked:  data/dmrs_single.RDS
    Untracked:  data/dmrs_single_15_06.RDS
    Untracked:  data/gene_count_matrix.csv
    Untracked:  data/gene_expression_mat.txt
    Untracked:  data/homer_results_merged.RDS
    Untracked:  data/motif_enrichment_AEL_subgroups.RDS
    Untracked:  data/mutations.txt
    Untracked:  data/ncomms9489-s3.xlsx
    Untracked:  data/ncomms9489-s7.xlsx
    Untracked:  data/no_snps_methrix_2023.RDS
    Untracked:  data/pcbc-stemsig.p219.Rda
    Untracked:  data/processed_rnbset_GSE63409.RDS
    Untracked:  data/raw_methrix_2023.RDS
    Untracked:  data/samples_stringtie.txt
    Untracked:  data/stringtie/
    Untracked:  data/transcript_count_matrix.csv
    Untracked:  data/tumor_vs._all_in_normal_down.txt
    Untracked:  data/tumor_vs._all_in_normal_up.txt
    Untracked:  data/tumor_vs._normal_in_CMP_cells_down.txt
    Untracked:  data/tumor_vs._normal_in_CMP_cells_up.txt
    Untracked:  data/tumor_vs._normal_in_HSC_cells_down.txt
    Untracked:  data/tumor_vs._normal_in_HSC_cells_up.txt
    Untracked:  data/tumor_vs._normal_in_MPP_cells_down.txt
    Untracked:  data/tumor_vs._normal_in_MPP_cells_up.txt
    Untracked:  omnipathr-log/
    Untracked:  output/filtered/
    Untracked:  output/filtered_MC_per_chr.tsv
    Untracked:  output/filtered_contig_lens.tsv
    Untracked:  output/filtered_global_MC_per_samp.tsv
    Untracked:  output/filtered_methrix_reports.html
    Untracked:  output/filtered_n_covered_by_all_samples.tsv
    Untracked:  output/filtered_n_covered_per_chr.tsv
    Untracked:  output/raw/
    Untracked:  output/raw_MC_per_chr.tsv
    Untracked:  output/raw_contig_lens.tsv
    Untracked:  output/raw_global_MC_per_samp.tsv
    Untracked:  output/raw_methrix_reports.html
    Untracked:  output/raw_n_covered_by_all_samples.tsv
    Untracked:  output/raw_n_covered_per_chr.tsv
    Untracked:  output/res_enhancer/

Unstaged changes:
    Modified:   analysis/DMR_calling.Rmd
    Modified:   analysis/DMR_report.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/differential_AML.Rmd</code>) and
HTML (<code>docs/differential_AML.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkik/AEL_methylation/blob/83a5ebf76ef26bb0e11b9b21d4606530b6df186e/analysis/differential_AML.Rmd" target="_blank">83a5ebf</a>
</td>
<td>
tkik
</td>
<td>
2023-06-16
</td>
<td>
wflow_publish(c("analysis/DMR_report_AEL_subgroups.Rmd",
"analysis/differential_AML.Rmd"))
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setting-folders-and-basic-parameters" class="section level2">
<h2>Setting folders and basic parameters</h2>
<pre class="r"><code>#Genome of your preference to work with
if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)

library(BiocManager)

if(!requireNamespace(&quot;BSgenome.Hsapiens.UCSC.hg19&quot;)) {
BiocManager::install(&quot;BSgenome.Hsapiens.UCSC.hg19&quot;)
}
library(BSgenome.Hsapiens.UCSC.hg19) 
library(annotatr)
library(ComplexHeatmap)
library(ChIPseeker)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(clusterProfiler)</code></pre>
<pre class="r"><code>meth &lt;- readRDS(file.path(&quot;data&quot;, &quot;no_snps_methrix_2023.RDS&quot;))
meth@colData &lt;- meth@colData[,-grep(&quot;X&quot;, colnames(meth@colData))]
meth@colData$Cell.Type &lt;- gsub(&quot; \\(.*\\)&quot;, &quot;&quot;, meth@colData$Cell.Type)
meth@colData$Group &lt;- as.character(meth@colData$Group)
meth@colData$Group[meth@colData$Group==&quot;&quot;] &lt;- &quot;Normal&quot;
  meth@colData$Cell.Type[is.na(meth@colData$Cell.Type)] &lt;- &quot;Bulk patient sample&quot;


dmrs  &lt;- readRDS(file=file.path(&quot;data&quot;, &quot;dmrs_single.RDS&quot;))
dmls &lt;- readRDS(file=file.path(&quot;data&quot;, &quot;dmls_single.RDS&quot;))

labels &lt;- strsplit(names(dmrs), split=&quot;_&quot;)
labels &lt;- lapply(labels, function(x) ifelse(length(x)==3, paste0(x[2], &quot; vs. &quot;, x[3], &quot; in &quot;, x[1], &quot; cells&quot;), paste0(x[1], &quot; vs. all in &quot;, x[2])))
labels &lt;- unlist(labels)
names(dmrs) &lt;- labels
names(dmls) &lt;- labels

#illumina &lt;- readRDS(&quot;data/processed_rnbset_GSE63409.RDS&quot;)
#illumina@pheno$ID &lt;- gsub(&quot;+&quot;, &quot;plus&quot;, illumina@pheno$ID, fixed = T)
#illumina@pheno$ID &lt;- gsub(&quot;([[:digit:]])-&quot;, &quot;\\1minus&quot;, illumina@pheno$ID )
#illumina@pheno$ID &lt;- gsub(&quot;-&quot;, &quot;.&quot;, illumina@pheno$ID, fixed = T)
#saveRDS(illumina, &quot;data/processed_rnbset_GSE63409.RDS&quot;)

illumina &lt;- readRDS(&quot;data/processed_rnbset_GSE63409.RDS&quot;)

illumina_anno &lt;- RnBeads::annotation(illumina)

meth2 &lt;- subset_methrix(meth, regions = makeGRangesFromDataFrame(illumina_anno))</code></pre>
<pre><code>-Subsetting by genomic regions</code></pre>
<pre class="r"><code>colnames(illumina_anno)[c(1,2,4)] &lt;- c(&quot;chr&quot;, &quot;start&quot;, &quot;strand&quot;)
 illumina_anno$strand &lt;- &quot;*&quot;
 coldata &lt;- pheno(illumina)
 rownames(coldata) &lt;- coldata$ID
 illumina_methrix &lt;- methrix:::create_methrix(beta_mat = meth(illumina), cov_mat = ceiling(meth(illumina)), cpg_loci = data.table(illumina_anno[,c(&quot;chr&quot;, &quot;start&quot;, &quot;strand&quot;)]),
is_hdf5 = FALSE, genome_name = &quot;hg19&quot;, col_data = pheno(illumina))
 meth2 &lt;- remove_uncovered(meth2)</code></pre>
<pre><code>-Removed 14 [0%] uncovered loci of 342,843 sites</code></pre>
<pre><code>-Finished in:  0.890s elapsed (0.810s cpu)</code></pre>
<pre class="r"><code> sds &lt;- rowSds(as.matrix(methrix::get_matrix(meth2)))
 meth2 &lt;- meth2[which(!is.na(sds)),]
 meth2 &lt;- meth2[order(sds[!is.na(sds)], decreasing = T)[1:30000],]
 
 
 overlapping &lt;- meth2@elementMetadata
 overlapping$end &lt;- overlapping$start+1
illumina_methrix &lt;- subset_methrix(illumina_methrix, makeGRangesFromDataFrame(overlapping))</code></pre>
<pre><code>-Subsetting by genomic regions</code></pre>
<pre class="r"><code>illumina_methrix@elementMetadata$chr &lt;- as.character(illumina_methrix@elementMetadata$chr)
illumina_methrix@elementMetadata$start &lt;- as.numeric(illumina_methrix@elementMetadata$start)
illumina_methrix@elementMetadata$strand &lt;- factor(illumina_methrix@elementMetadata$strand, levels = c(&quot;+&quot;, &quot;-&quot;, &quot;*&quot;))

plot_anno &lt;- as.data.frame(meth@colData)[,c(&quot;Sample.Type&quot;, &quot;Cell.Type&quot;, &quot;Group&quot;)]
plot_anno$Sample.Type2 &lt;- ifelse(plot_anno$Sample.Type==&quot;NORMAL&quot;, &quot;normal&quot;, &quot;AEL&quot;)
plot_anno$Cell.Type &lt;- gsub(&quot;sorted &quot;, &quot;&quot;, plot_anno$Cell.Type)
plot_anno$origin &lt;- &quot;Seq&quot;
#plot_anno$origin[plot_anno$Sample.Type2!=&quot;normal&quot;] &lt;- NA
meth2@colData &lt;- DataFrame(plot_anno[,c(&quot;Cell.Type&quot;, &quot;Sample.Type2&quot;,&quot;Group&quot;,  &quot;origin&quot;)])



plot_anno_illumina &lt;- as.data.frame(illumina_methrix@colData)
plot_anno_illumina &lt;- plot_anno_illumina[,c(&quot;phenotype.ch1&quot;, &quot;subject_status.ch1&quot;)]
plot_anno_illumina$phenotype.ch1 &lt;- gsub(&quot; \\(.+\\)&quot;, &quot;&quot;, plot_anno_illumina$phenotype.ch1)
colnames(plot_anno_illumina) &lt;- c(&quot;Cell.Type&quot;, &quot;Sample.Type2&quot;)
plot_anno_illumina$Group &lt;- NA
plot_anno_illumina$origin &lt;- &quot;ilumina&quot;
#plot_anno_illumina$origin[plot_anno_illumina$Sample.Type2!=&quot;normal&quot;] &lt;- NA
illumina_methrix@colData &lt;- DataFrame(plot_anno_illumina)


illumina_methrix@elementMetadata &lt;- illumina_methrix@elementMetadata[order(illumina_methrix@elementMetadata$chr),] 
meth2@elementMetadata &lt;- meth2@elementMetadata[order(meth2@elementMetadata$chr, meth2@elementMetadata$start),] 

combined_data &lt;- methrix::combine_methrix(illumina_methrix, meth2, by=&quot;col&quot;)



mat &lt;- methrix::get_matrix(combined_data)
rownames(mat) &lt;- paste0(combined_data@elementMetadata$chr, &quot;_&quot;, combined_data@elementMetadata$start)

design &lt;- model.matrix(~0+Sample.Type2, data = combined_data@colData)
contrast.matrix &lt;- makeContrasts(Sample.Type2AEL-Sample.Type2normal, Sample.Type2AEL-Sample.Type2AML, Sample.Type2AML-Sample.Type2normal, levels=design)
param &lt;- SnowParam(workers = 8, type = &quot;SOCK&quot;)

mat &lt;- mat[complete.cases(mat),]
#mat2 &lt;- mat[sample(1:nrow(mat), 20000),]
set.seed(234)
pheatmap(mat[sample(1:nrow(mat), 5000),], annotation_col = as.data.frame(combined_data@colData), show_rownames = F, show_colnames = F)</code></pre>
<pre><code>`use_raster` is automatically set to TRUE for a matrix with more than
2000 rows. You can control `use_raster` argument by explicitly setting
TRUE/FALSE to it.

Set `ht_opt$message = FALSE` to turn off this message.</code></pre>
<p><img src="figure/differential_AML.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>mat_combat &lt;- ComBat(as.matrix(mat), combined_data$origin)</code></pre>
<pre><code>Found2batches</code></pre>
<pre><code>Adjusting for0covariate(s) or covariate level(s)</code></pre>
<pre><code>Standardizing Data across genes</code></pre>
<pre><code>Fitting L/S model and finding priors</code></pre>
<pre><code>Finding parametric adjustments</code></pre>
<pre><code>Adjusting the Data</code></pre>
<pre class="r"><code>set.seed(234)
pheatmap(mat_combat[sample(1:nrow(mat_combat), 5000),], annotation_col = as.data.frame(combined_data@colData), show_rownames = F, show_colnames = F)</code></pre>
<pre><code>`use_raster` is automatically set to TRUE for a matrix with more than
2000 rows. You can control `use_raster` argument by explicitly setting
TRUE/FALSE to it.

Set `ht_opt$message = FALSE` to turn off this message.</code></pre>
<p><img src="figure/differential_AML.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>fit &lt;- lmFit(mat_combat, design)
fit2 &lt;- contrasts.fit(fit, contrast.matrix)
fit2 &lt;- eBayes(fit2)

res_AEL_normal &lt;- topTable(fit2, coef=1, adjust=&quot;BH&quot;, number = 20000)
pheatmap(mat_combat[rownames(res_AEL_normal[res_AEL_normal$adj.P.Val&lt;0.05,]),], annotation_col = as.data.frame(combined_data@colData), show_rownames = F, show_colnames = F)</code></pre>
<pre><code>`use_raster` is automatically set to TRUE for a matrix with more than
2000 rows. You can control `use_raster` argument by explicitly setting
TRUE/FALSE to it.

Set `ht_opt$message = FALSE` to turn off this message.</code></pre>
<p><img src="figure/differential_AML.Rmd/unnamed-chunk-2-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res_AEL_AML &lt;- topTable(fit2, coef=2, adjust=&quot;BH&quot;, number = 20000)
pheatmap(mat_combat[rownames(res_AEL_AML[res_AEL_AML$adj.P.Val&lt;0.05,]),], annotation_col = as.data.frame(combined_data@colData), show_rownames = F, show_colnames = F)</code></pre>
<pre><code>`use_raster` is automatically set to TRUE for a matrix with more than
2000 rows. You can control `use_raster` argument by explicitly setting
TRUE/FALSE to it.

Set `ht_opt$message = FALSE` to turn off this message.</code></pre>
<p><img src="figure/differential_AML.Rmd/unnamed-chunk-2-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sig_mat &lt;- mat[intersect(rownames(res1[res1$adj.P.Val&lt;0.0001 &amp; abs(res1$logFC)&gt;0.5,]),rownames(results@.Data[results@.Data[,1]!=0,])),  ]


  pheatmap::pheatmap(sig_mat, annotation_col = as.data.frame(combined_data@colData), colorRampPalette(Cellpaper, space = &quot;rgb&quot;)(10), main = comp, show_rownames = FALSE, show_colnames = F)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.3 (2023-03-15 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] RnBeads.hg19_1.30.0                    
 [2] clusterProfiler_4.6.0                  
 [3] ChIPseeker_1.34.1                      
 [4] ComplexHeatmap_2.14.0                  
 [5] annotatr_1.24.0                        
 [6] BSgenome.Hsapiens.UCSC.hg19_1.4.3      
 [7] BSgenome_1.66.3                        
 [8] rtracklayer_1.58.0                     
 [9] BiocManager_1.30.20                    
[10] sva_3.46.0                             
[11] genefilter_1.80.3                      
[12] mgcv_1.8-42                            
[13] nlme_3.1-162                           
[14] RnBeads_2.16.0                         
[15] plyr_1.8.8                             
[16] methylumi_2.44.0                       
[17] minfi_1.44.0                           
[18] bumphunter_1.40.0                      
[19] locfit_1.5-9.7                         
[20] iterators_1.0.14                       
[21] foreach_1.5.2                          
[22] Biostrings_2.66.0                      
[23] XVector_0.38.0                         
[24] FDb.InfiniumMethylation.hg19_2.2.0     
[25] org.Hs.eg.db_3.16.0                    
[26] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
[27] GenomicFeatures_1.50.4                 
[28] AnnotationDbi_1.60.0                   
[29] reshape2_1.4.4                         
[30] scales_1.2.1                           
[31] illuminaio_0.40.0                      
[32] limma_3.54.1                           
[33] gridExtra_2.3                          
[34] gplots_3.1.3                           
[35] fields_14.1                            
[36] viridis_0.6.2                          
[37] viridisLite_0.4.1                      
[38] spam_2.9-1                             
[39] ff_4.0.9                               
[40] bit_4.0.5                              
[41] cluster_2.1.4                          
[42] MASS_7.3-58.2                          
[43] plotly_4.10.1                          
[44] ggplot2_3.4.1                          
[45] DSS_2.46.0                             
[46] bsseq_1.34.0                           
[47] BiocParallel_1.32.5                    
[48] pheatmap_1.0.12                        
[49] methrix_1.8.01                         
[50] SummarizedExperiment_1.28.0            
[51] Biobase_2.58.0                         
[52] GenomicRanges_1.50.2                   
[53] GenomeInfoDb_1.34.9                    
[54] IRanges_2.32.0                         
[55] S4Vectors_0.36.1                       
[56] BiocGenerics_0.44.0                    
[57] MatrixGenerics_1.10.0                  
[58] matrixStats_0.63.0                     
[59] data.table_1.14.8                      
[60] workflowr_1.7.0.4                      

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3                R.methodsS3_1.8.2            
  [3] tidyr_1.3.0                   bit64_4.0.5                  
  [5] knitr_1.42                    DelayedArray_0.23.2          
  [7] R.utils_2.12.2                KEGGREST_1.38.0              
  [9] RCurl_1.98-1.10               GEOquery_2.66.0              
 [11] doParallel_1.0.17             generics_0.1.3               
 [13] preprocessCore_1.60.2         cowplot_1.1.1                
 [15] callr_3.7.3                   RSQLite_2.3.0                
 [17] shadowtext_0.1.2              enrichplot_1.18.3            
 [19] tzdb_0.3.0                    xml2_1.3.3                   
 [21] httpuv_1.6.6                  assertthat_0.2.1             
 [23] xfun_0.37                     hms_1.1.2                    
 [25] jquerylib_0.1.4               evaluate_0.20                
 [27] promises_1.2.0.1              fansi_1.0.4                  
 [29] restfulr_0.0.15               scrime_1.3.5                 
 [31] progress_1.2.2                caTools_1.18.2               
 [33] dbplyr_2.3.0                  igraph_1.4.0                 
 [35] DBI_1.1.3                     htmlwidgets_1.6.1            
 [37] reshape_0.8.9                 purrr_1.0.1                  
 [39] ellipsis_0.3.2                dplyr_1.1.0                  
 [41] permute_0.9-7                 annotate_1.76.0              
 [43] biomaRt_2.54.0                sparseMatrixStats_1.10.0     
 [45] vctrs_0.5.2                   Cairo_1.6-0                  
 [47] cachem_1.0.6                  withr_2.5.0                  
 [49] ggforce_0.4.1                 HDO.db_0.99.1                
 [51] treeio_1.22.0                 GenomicAlignments_1.34.0     
 [53] prettyunits_1.1.1             mclust_6.0.0                 
 [55] DOSE_3.24.2                   ape_5.7                      
 [57] dotCall64_1.0-2               lazyeval_0.2.2               
 [59] crayon_1.5.2                  edgeR_3.40.2                 
 [61] pkgconfig_2.0.3               tweenr_2.0.2                 
 [63] rlang_1.0.6                   lifecycle_1.0.3              
 [65] downloader_0.4                filelock_1.0.2               
 [67] BiocFileCache_2.6.1           AnnotationHub_3.6.0          
 [69] polyclip_1.10-4               rprojroot_2.0.3              
 [71] rngtools_1.5.2                aplot_0.1.9                  
 [73] base64_2.0.1                  Matrix_1.5-3                 
 [75] boot_1.3-28.1                 Rhdf5lib_1.20.0              
 [77] whisker_0.4.1                 GlobalOptions_0.1.2          
 [79] processx_3.7.0                png_0.1-8                    
 [81] rjson_0.2.21                  bitops_1.0-7                 
 [83] gson_0.0.9                    getPass_0.2-2                
 [85] R.oo_1.25.0                   KernSmooth_2.23-20           
 [87] rhdf5filters_1.10.0           blob_1.2.3                   
 [89] DelayedMatrixStats_1.20.0     shape_1.4.6                  
 [91] doRNG_1.8.6                   qvalue_2.30.0                
 [93] stringr_1.5.0                 nor1mix_1.3-0                
 [95] regioneR_1.30.0               gridGraphics_0.5-1           
 [97] readr_2.1.4                   memoise_2.0.1                
 [99] magrittr_2.0.3                zlibbioc_1.44.0              
[101] scatterpie_0.1.8              compiler_4.2.3               
[103] BiocIO_1.8.0                  RColorBrewer_1.1-3           
[105] plotrix_3.8-2                 clue_0.3-64                  
[107] Rsamtools_2.14.0              cli_3.6.0                    
[109] patchwork_1.1.2               ps_1.7.0                     
[111] tidyselect_1.2.0              stringi_1.7.8                
[113] highr_0.10                    GOSemSim_2.24.0              
[115] yaml_2.3.6                    askpass_1.1                  
[117] ggrepel_0.9.3                 sass_0.4.5                   
[119] fastmatch_1.1-3               tools_4.2.3                  
[121] circlize_0.4.15               rstudioapi_0.14              
[123] git2r_0.30.1                  farver_2.1.1                 
[125] ggraph_2.1.0                  digest_0.6.29                
[127] shiny_1.7.4                   quadprog_1.5-8               
[129] Rcpp_1.0.9                    siggenes_1.72.0              
[131] BiocVersion_3.16.0            later_1.3.0                  
[133] httr_1.4.4                    colorspace_2.1-0             
[135] XML_3.99-0.13                 fs_1.5.2                     
[137] splines_4.2.3                 yulab.utils_0.0.6            
[139] tidytree_0.4.2                graphlayouts_0.8.4           
[141] multtest_2.54.0               ggplotify_0.1.0              
[143] xtable_1.8-4                  ggtree_3.6.2                 
[145] jsonlite_1.8.4                tidygraph_1.2.3              
[147] ggfun_0.0.9                   R6_2.5.1                     
[149] pillar_1.8.1                  htmltools_0.5.4              
[151] mime_0.12                     glue_1.6.2                   
[153] fastmap_1.1.0                 interactiveDisplayBase_1.36.0
[155] beanplot_1.3.1                codetools_0.2-19             
[157] fgsea_1.24.0                  maps_3.4.1                   
[159] utf8_1.2.3                    lattice_0.20-45              
[161] bslib_0.4.2                   tibble_3.1.8                 
[163] curl_5.0.0                    gtools_3.9.4                 
[165] magick_2.7.3                  GO.db_3.16.0                 
[167] openssl_2.0.5                 survival_3.5-3               
[169] rmarkdown_2.20                munsell_0.5.0                
[171] GetoptLong_1.0.5              rhdf5_2.42.0                 
[173] GenomeInfoDbData_1.2.9        HDF5Array_1.26.0             
[175] gtable_0.3.1                 </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
