<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Reka_Toth" />


<title>DMR report</title>

<script src="site_libs/header-attrs-2.17/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.10.0/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-2.5.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-2.5.1/plotly-latest.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">22_AEL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">DMR report</h1>
<h4 class="author">Reka_Toth</h4>
<h4 class="date">03 February, 2023</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-02-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>22_AEL/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230201code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230201)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230201code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230201)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtkikAELmethylationtreed2e0b8f9b74afd8e0c8e602eeca7368e957040dbtargetblankd2e0b8fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tkik/AEL_methylation/tree/d2e0b8f9b74afd8e0c8e602eeca7368e957040db" target="_blank">d2e0b8f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtkikAELmethylationtreed2e0b8f9b74afd8e0c8e602eeca7368e957040dbtargetblankd2e0b8fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tkik/AEL_methylation/tree/d2e0b8f9b74afd8e0c8e602eeca7368e957040db" target="_blank">d2e0b8f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    refdir/

Untracked files:
    Untracked:  analysis/01_read_in.Rmd
    Untracked:  analysis/3_combine_meth_ge_mofa.Rmd
    Untracked:  analysis/candidate_genes.Rmd
    Untracked:  analysis/enhancer_gene_pairs.Rmd
    Untracked:  analysis/epiCMIT_score.Rmd
    Untracked:  analysis/homer_all.Rmd
    Untracked:  analysis/homer_results_promoter.Rmd
    Untracked:  analysis/inital_analysis.Rmd
    Untracked:  analysis/prepare_homer.Rmd
    Untracked:  analysis/test.Rmd
    Untracked:  code/dataETOC2.Rd
    Untracked:  code/epiTOC2.R
    Untracked:  code/homer_analysis_2023_01.bash
    Untracked:  code/region_annotation_function.R
    Untracked:  code/seq2gene_func.R
    Untracked:  data/CMP_vs._all_in_normal_down.txt
    Untracked:  data/CMP_vs._all_in_normal_up.txt
    Untracked:  data/CMP_vs._all_in_tumor_down.txt
    Untracked:  data/CMP_vs._all_in_tumor_up.txt
    Untracked:  data/DML_tests.RDS
    Untracked:  data/HSC_vs._all_in_normal_down.txt
    Untracked:  data/HSC_vs._all_in_normal_up.txt
    Untracked:  data/HSC_vs._all_in_tumor_down.txt
    Untracked:  data/HSC_vs._all_in_tumor_up.txt
    Untracked:  data/MPP_vs._all_in_normal_down.txt
    Untracked:  data/MPP_vs._all_in_normal_up.txt
    Untracked:  data/MPP_vs._all_in_tumor_down.txt
    Untracked:  data/MPP_vs._all_in_tumor_up.txt
    Untracked:  data/deseq.RDS
    Untracked:  data/dmls_single.RDS
    Untracked:  data/dmr_beds/
    Untracked:  data/dmr_beds_enhancer/
    Untracked:  data/dmr_beds_promoter/
    Untracked:  data/dmrs_single.RDS
    Untracked:  data/gene_count_matrix.csv
    Untracked:  data/homer_results_merged.RDS
    Untracked:  data/mutations.txt
    Untracked:  data/no_snps_methrix_2023.RDS
    Untracked:  data/pcbc-stemsig.p219.Rda
    Untracked:  data/raw_methrix_2023.RDS
    Untracked:  data/samples_stringtie.txt
    Untracked:  data/transcript_count_matrix.csv
    Untracked:  data/tumor_vs._all_in_normal_down.txt
    Untracked:  data/tumor_vs._all_in_normal_up.txt
    Untracked:  data/tumor_vs._normal_in_CMP_cells_down.txt
    Untracked:  data/tumor_vs._normal_in_CMP_cells_up.txt
    Untracked:  data/tumor_vs._normal_in_HSC_cells_down.txt
    Untracked:  data/tumor_vs._normal_in_HSC_cells_up.txt
    Untracked:  data/tumor_vs._normal_in_MPP_cells_down.txt
    Untracked:  data/tumor_vs._normal_in_MPP_cells_up.txt
    Untracked:  dataCMP_vs._all_in_normal_down.txt
    Untracked:  dataCMP_vs._all_in_normal_up.txt
    Untracked:  dataCMP_vs._all_in_tumor_down.txt
    Untracked:  dataCMP_vs._all_in_tumor_up.txt
    Untracked:  dataHSC_vs._all_in_normal_down.txt
    Untracked:  dataHSC_vs._all_in_normal_up.txt
    Untracked:  dataHSC_vs._all_in_tumor_down.txt
    Untracked:  dataHSC_vs._all_in_tumor_up.txt
    Untracked:  dataMPP_vs._all_in_normal_down.txt
    Untracked:  dataMPP_vs._all_in_normal_up.txt
    Untracked:  dataMPP_vs._all_in_tumor_down.txt
    Untracked:  dataMPP_vs._all_in_tumor_up.txt
    Untracked:  datatumor_vs._normal_in_CMP_cells_down.txt
    Untracked:  datatumor_vs._normal_in_CMP_cells_up.txt
    Untracked:  datatumor_vs._normal_in_HSC_cells_down.txt
    Untracked:  datatumor_vs._normal_in_HSC_cells_up.txt
    Untracked:  datatumor_vs._normal_in_MPP_cells_down.txt
    Untracked:  datatumor_vs._normal_in_MPP_cells_up.txt
    Untracked:  output/MDS.html
    Untracked:  output/MDS_files/
    Untracked:  output/MDS_full.html
    Untracked:  output/MDS_full_files/
    Untracked:  output/TFBS.pdf
    Untracked:  output/filtered/
    Untracked:  output/filtered_MC_per_chr.tsv
    Untracked:  output/filtered_contig_lens.tsv
    Untracked:  output/filtered_global_MC_per_samp.tsv
    Untracked:  output/filtered_methrix_reports.html
    Untracked:  output/filtered_n_covered_by_all_samples.tsv
    Untracked:  output/filtered_n_covered_per_chr.tsv
    Untracked:  output/raw/
    Untracked:  output/raw_MC_per_chr.tsv
    Untracked:  output/raw_contig_lens.tsv
    Untracked:  output/raw_global_MC_per_samp.tsv
    Untracked:  output/raw_methrix_reports.html
    Untracked:  output/raw_n_covered_by_all_samples.tsv
    Untracked:  output/raw_n_covered_per_chr.tsv
    Untracked:  output/res_enhancer/

Unstaged changes:
    Modified:   analysis/1_read_in_RNA_seq.Rmd
    Modified:   analysis/DMR_calling.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/DMR_report.Rmd</code>) and HTML
(<code>docs/DMR_report.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkik/AEL_methylation/blob/d2e0b8f9b74afd8e0c8e602eeca7368e957040db/analysis/DMR_report.Rmd" target="_blank">d2e0b8f</a>
</td>
<td>
tkik
</td>
<td>
2023-02-03
</td>
<td>
wflow_publish(files = c("analysis/epiCMIT_score_comb.Rmd",
"analysis/DMR_report.Rmd",
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkik/AEL_methylation/blob/f155798f94cf8ad49699b983765ab485e0d465eb/analysis/DMR_report.Rmd" target="_blank">f155798</a>
</td>
<td>
tkik
</td>
<td>
2023-02-01
</td>
<td>
initial analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setting-folders-and-basic-parameters" class="section level2">
<h2>Setting folders and basic parameters</h2>
<pre class="r"><code>#Genome of your preference to work with
if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)

library(BiocManager)

if(!requireNamespace(&quot;BSgenome.Hsapiens.UCSC.hg19&quot;)) {
BiocManager::install(&quot;BSgenome.Hsapiens.UCSC.hg19&quot;)
}
library(BSgenome.Hsapiens.UCSC.hg19) 
library(annotatr)
library(ComplexHeatmap)
library(ChIPseeker)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(clusterProfiler)</code></pre>
<pre class="r"><code>meth &lt;- readRDS(file.path(&quot;data&quot;, &quot;no_snps_methrix_2023.RDS&quot;))
meth@colData &lt;- meth@colData[,-grep(&quot;X&quot;, colnames(meth@colData))]
meth@colData$Cell.Type &lt;- gsub(&quot; \\(.*\\)&quot;, &quot;&quot;, meth@colData$Cell.Type)
meth@colData$Group &lt;- as.character(meth@colData$Group)
meth@colData$Group[meth@colData$Group==&quot;&quot;] &lt;- &quot;Normal&quot;
  meth@colData$Cell.Type[is.na(meth@colData$Cell.Type)] &lt;- &quot;Bulk patient sample&quot;


dmrs  &lt;- readRDS(file=file.path(&quot;data&quot;, &quot;dmrs_single.RDS&quot;))
dmls &lt;- readRDS(file=file.path(&quot;data&quot;, &quot;dmls_single.RDS&quot;))

labels &lt;- strsplit(names(dmrs), split=&quot;_&quot;)
labels &lt;- lapply(labels, function(x) ifelse(length(x)==3, paste0(x[2], &quot; vs. &quot;, x[3], &quot; in &quot;, x[1], &quot; cells&quot;), paste0(x[1], &quot; vs. all in &quot;, x[2])))
labels &lt;- unlist(labels)
names(dmrs) &lt;- labels
names(dmls) &lt;- labels</code></pre>
</div>
<div id="general-statistics" class="section level1">
<h1>General statistics</h1>
<div id="distribution-and-overlap-of-differentially-methylated-sites"
class="section level2">
<h2>Distribution and overlap of differentially methylated sites</h2>
<p>The upset plot shows the number of differentially methylated sites
and their overlap in the different categories.</p>
<pre class="r"><code>#create a dataframe with all the sites in the methylation data
all_sites &lt;- setDT(as.data.frame(meth@elementMetadata))

#add an end column to the dataframe
all_sites$end &lt;- all_sites$start+2

#set the key for the dataframe
setkeyv(all_sites, cols = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;))

#for each comparison in the list of comparisons
for (comp in names(dmls)){
  
  #rename the first two columns of the dataframe to chr and start
  colnames(dmls[[comp]])[1:2] &lt;- c(&quot;chr&quot;, &quot;start&quot;)
  
  #convert the dataframe to a data table
  setDT(dmls[[comp]])
  
  #add an end column to the data table
  dmls[[comp]][,end:=start+2]
  
  #set the key for the data table
  setkeyv(dmls[[comp]], cols = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;))
  
  #find the overlapping sites between the data table and the dataframe
  overlapping &lt;- data.table::foverlaps(all_sites, dmls[[comp]], type = &quot;start&quot;, nomatch = NA, which = TRUE, by.x = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;), by.y = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;))$yid
  
  #add a column to the dataframe with the overlapping sites
  all_sites[,comp:=overlapping]
  
  #add a column to the dataframe with a 1 if the site is overlapping and a 0 if it is not
  all_sites[,eval(comp):=ifelse(is.na(all_sites[,comp]), 0,1)]
  }

#remove the column with the overlapping sites
all_sites[,comp:=NULL]
UpSetR::upset(all_sites, sets = names(dmls))</code></pre>
<p><img src="figure/DMR_report.Rmd/upset%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="number-and-distribution-of-differentially-methylated-sites"
class="section level2">
<h2>Number and distribution of differentially methylated sites</h2>
<div id="htmlwidget-a1b38d5fe79e422599b4" style="width:672px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-a1b38d5fe79e422599b4">{"x":{"data":[{"orientation":"v","width":[0.899999999999999,0.9,0.9,0.899999999999999,0.9,0.9,0.899999999999999,0.9,0.9,0.9],"base":[11.3066313617125,6.64385618977472,5.93073733756289,11.0525680508042,4.32192809488736,8.75822321472672,11.9610878899973,9.45532722030456,4.90689059560852,18.8751493834161],"x":[8,2,1,9,4,3,10,6,5,7],"y":[16.6862132508221,0,9.88874324889826,18.1786162261361,7.65821148275179,4.90689059560852,17.6593130588029,6.74146698640115,3.70043971814109,12.3763967510828],"text":["Comp: tumor vs. normal in CMP cells<br />Count: 16.686213<br />Direction: Gain in second","Comp: CMP vs. all in tumor<br />Count:  0.000000<br />Direction: Gain in second","Comp: CMP vs. all in normal<br />Count:  9.888743<br />Direction: Gain in second","Comp: tumor vs. normal in HSC cells<br />Count: 18.178616<br />Direction: Gain in second","Comp: HSC vs. all in tumor<br />Count:  7.658211<br />Direction: Gain in second","Comp: HSC vs. all in normal<br />Count:  4.906891<br />Direction: Gain in second","Comp: tumor vs. normal in MPP cells<br />Count: 17.659313<br />Direction: Gain in second","Comp: MPP vs. all in tumor<br />Count:  6.741467<br />Direction: Gain in second","Comp: MPP vs. all in normal<br />Count:  3.700440<br />Direction: Gain in second","Comp: tumor vs. all in normal<br />Count: 12.376397<br />Direction: Gain in second"],"type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(27,158,119,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Gain in second","legendgroup":"Gain in second","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.899999999999999,0.9,0.9,0.899999999999999,0.9,0.9,0.899999999999999,0.9,0.9,0.9],"base":[0,0,0,0,0,0,0,0,0,0],"x":[8,2,1,9,4,3,10,6,5,7],"y":[11.3066313617125,6.64385618977472,5.93073733756289,11.0525680508042,4.32192809488736,8.75822321472672,11.9610878899973,9.45532722030456,4.90689059560852,18.8751493834161],"text":["Comp: tumor vs. normal in CMP cells<br />Count: 11.306631<br />Direction: Loss in second","Comp: CMP vs. all in tumor<br />Count:  6.643856<br />Direction: Loss in second","Comp: CMP vs. all in normal<br />Count:  5.930737<br />Direction: Loss in second","Comp: tumor vs. normal in HSC cells<br />Count: 11.052568<br />Direction: Loss in second","Comp: HSC vs. all in tumor<br />Count:  4.321928<br />Direction: Loss in second","Comp: HSC vs. all in normal<br />Count:  8.758223<br />Direction: Loss in second","Comp: tumor vs. normal in MPP cells<br />Count: 11.961088<br />Direction: Loss in second","Comp: MPP vs. all in tumor<br />Count:  9.455327<br />Direction: Loss in second","Comp: MPP vs. all in normal<br />Count:  4.906891<br />Direction: Loss in second","Comp: tumor vs. all in normal<br />Count: 18.875149<br />Direction: Loss in second"],"type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(217,95,2,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Loss in second","legendgroup":"Loss in second","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":26.2283105022831,"r":7.30593607305936,"b":103.13365509806,"l":37.2602739726027},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.4,10.6],"tickmode":"array","ticktext":["CMP vs. all in normal","CMP vs. all in tumor","HSC vs. all in normal","HSC vs. all in tumor","MPP vs. all in normal","MPP vs. all in tumor","tumor vs. all in normal","tumor vs. normal in CMP cells","tumor vs. normal in HSC cells","tumor vs. normal in MPP cells"],"tickvals":[1,2,3,4,5,6,7,8,9,10],"categoryorder":"array","categoryarray":["CMP vs. all in normal","CMP vs. all in tumor","HSC vs. all in normal","HSC vs. all in tumor","MPP vs. all in normal","MPP vs. all in tumor","tumor vs. all in normal","tumor vs. normal in CMP cells","tumor vs. normal in HSC cells","tumor vs. normal in MPP cells"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-45,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"y","title":{"text":"","font":{"color":null,"family":null,"size":0}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1.56257730672495,32.8141234412239],"tickmode":"array","ticktext":["0","10","20","30"],"tickvals":[0,10,20,30],"categoryorder":"array","categoryarray":["0","10","20","30"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"x","title":{"text":"Log2(Count)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"title":{"text":"Direction","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"7a7c630e7886":{"x":{},"y":{},"fill":{},"type":"bar"}},"cur_data":"7a7c630e7886","visdat":{"7a7c630e7886":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-6547f64dcb9cc0f0b6eb" style="width:672px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-6547f64dcb9cc0f0b6eb">{"x":{"data":[{"orientation":"v","width":[0.899999999999999,0.9,0.9,0.899999999999999,0.9,0.9,0.899999999999999,0.9,0.9,0.9],"base":[20.026477514671,13.961901822341,14.1750816748283,19.6969861980419,11.6013065261218,16.7786803514548,20.6048572332129,17.0304300582502,12.3032099489448,29.1938632634579],"x":[8,2,1,9,4,3,10,6,5,7],"y":[25.6867531292149,8.66888498426625,17.9960375866926,27.4333559415841,15.2024299660153,12.7161764058074,26.8669901394247,14.349419464341,10.8719052376592,21.3402135920859],"text":["Comp: tumor vs. normal in CMP cells<br />Length: 25.686753<br />Direction: Gain in second","Comp: CMP vs. all in tumor<br />Length:  8.668885<br />Direction: Gain in second","Comp: CMP vs. all in normal<br />Length: 17.996038<br />Direction: Gain in second","Comp: tumor vs. normal in HSC cells<br />Length: 27.433356<br />Direction: Gain in second","Comp: HSC vs. all in tumor<br />Length: 15.202430<br />Direction: Gain in second","Comp: HSC vs. all in normal<br />Length: 12.716176<br />Direction: Gain in second","Comp: tumor vs. normal in MPP cells<br />Length: 26.866990<br />Direction: Gain in second","Comp: MPP vs. all in tumor<br />Length: 14.349419<br />Direction: Gain in second","Comp: MPP vs. all in normal<br />Length: 10.871905<br />Direction: Gain in second","Comp: tumor vs. all in normal<br />Length: 21.340214<br />Direction: Gain in second"],"type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(27,158,119,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Gain in second","legendgroup":"Gain in second","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.899999999999999,0.9,0.9,0.899999999999999,0.9,0.9,0.899999999999999,0.9,0.9,0.9],"base":[0,0,0,0,0,0,0,0,0,0],"x":[8,2,1,9,4,3,10,6,5,7],"y":[20.026477514671,13.961901822341,14.1750816748283,19.6969861980419,11.6013065261218,16.7786803514548,20.6048572332129,17.0304300582502,12.3032099489448,29.1938632634579],"text":["Comp: tumor vs. normal in CMP cells<br />Length: 20.026478<br />Direction: Loss in second","Comp: CMP vs. all in tumor<br />Length: 13.961902<br />Direction: Loss in second","Comp: CMP vs. all in normal<br />Length: 14.175082<br />Direction: Loss in second","Comp: tumor vs. normal in HSC cells<br />Length: 19.696986<br />Direction: Loss in second","Comp: HSC vs. all in tumor<br />Length: 11.601307<br />Direction: Loss in second","Comp: HSC vs. all in normal<br />Length: 16.778680<br />Direction: Loss in second","Comp: tumor vs. normal in MPP cells<br />Length: 20.604857<br />Direction: Loss in second","Comp: MPP vs. all in tumor<br />Length: 17.030430<br />Direction: Loss in second","Comp: MPP vs. all in normal<br />Length: 12.303210<br />Direction: Loss in second","Comp: tumor vs. all in normal<br />Length: 29.193863<br />Direction: Loss in second"],"type":"bar","textposition":"none","marker":{"autocolorscale":false,"color":"rgba(217,95,2,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Loss in second","legendgroup":"Loss in second","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":26.2283105022831,"r":7.30593607305936,"b":103.13365509806,"l":37.2602739726027},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.4,10.6],"tickmode":"array","ticktext":["CMP vs. all in normal","CMP vs. all in tumor","HSC vs. all in normal","HSC vs. all in tumor","MPP vs. all in normal","MPP vs. all in tumor","tumor vs. all in normal","tumor vs. normal in CMP cells","tumor vs. normal in HSC cells","tumor vs. normal in MPP cells"],"tickvals":[1,2,3,4,5,6,7,8,9,10],"categoryorder":"array","categoryarray":["CMP vs. all in normal","CMP vs. all in tumor","HSC vs. all in normal","HSC vs. all in tumor","MPP vs. all in normal","MPP vs. all in tumor","tumor vs. all in normal","tumor vs. normal in CMP cells","tumor vs. normal in HSC cells","tumor vs. normal in MPP cells"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-45,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"y","title":{"text":"","font":{"color":null,"family":null,"size":0}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-2.52670384277719,53.060780698321],"tickmode":"array","ticktext":["0","10","20","30","40","50"],"tickvals":[0,10,20,30,40,50],"categoryorder":"array","categoryarray":["0","10","20","30","40","50"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"x","title":{"text":"Log2(Length)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"title":{"text":"Direction","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"7a7c1b3c128c":{"x":{},"y":{},"fill":{},"type":"bar"}},"cur_data":"7a7c1b3c128c","visdat":{"7a7c1b3c128c":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="heatmap-of-dmrs" class="section level2 tabset tabset-fade">
<h2 class="tabset tabset-fade">Heatmap of DMRs</h2>
<pre class="r"><code>for (comp in names(dmrs)) {


 mat1 &lt;- as.data.frame(methrix::get_region_summary(meth, makeGRangesFromDataFrame(dmrs[[comp]][order(abs(dmrs[[comp]]$diff.Methy), decreasing = T)[1:min(nrow(dmrs[[comp]]), 20000)],])))
 
  plot_mat &lt;- mat1[complete.cases(mat1),-(1:5)]
  
  #p &lt;- pheatmap::pheatmap(plot_mat, annotation_col = as.data.frame(meth@colData)[,c(&quot;Sample.Type&quot;, &quot;Cell.Type&quot;)], colorRampPalette(Cellpaper, space = &quot;rgb&quot;)(10), main = comp, show_rownames = FALSE)
   
  cat(&quot;\n&quot;)
  cat(&quot;### Comparisons&quot;,
      comp,
      &quot;\n&quot;)
  pheatmap::pheatmap(plot_mat, annotation_col = as.data.frame(meth@colData)[,c(&quot;Sample.Type&quot;, &quot;Cell.Type&quot;)], colorRampPalette(Cellpaper, space = &quot;rgb&quot;)(10), main = comp, show_rownames = FALSE)
  
   cat(&quot;\n&quot;)
}</code></pre>
<div id="comparisons-tumor-vs.-normal-in-cmp-cells"
class="section level3">
<h3>Comparisons tumor vs. normal in CMP cells</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-cmp-vs.-all-in-tumor" class="section level3">
<h3>Comparisons CMP vs. all in tumor</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-cmp-vs.-all-in-normal" class="section level3">
<h3>Comparisons CMP vs. all in normal</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-tumor-vs.-normal-in-hsc-cells"
class="section level3">
<h3>Comparisons tumor vs. normal in HSC cells</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-hsc-vs.-all-in-tumor" class="section level3">
<h3>Comparisons HSC vs. all in tumor</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-hsc-vs.-all-in-normal" class="section level3">
<h3>Comparisons HSC vs. all in normal</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps-6.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-tumor-vs.-normal-in-mpp-cells"
class="section level3">
<h3>Comparisons tumor vs. normal in MPP cells</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps-7.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-mpp-vs.-all-in-tumor" class="section level3">
<h3>Comparisons MPP vs. all in tumor</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-mpp-vs.-all-in-normal" class="section level3">
<h3>Comparisons MPP vs. all in normal</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps-9.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-tumor-vs.-all-in-normal" class="section level3">
<h3>Comparisons tumor vs. all in normal</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps-10.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="heatmap-of-dmrs-overlap-with-promoters"
class="section level2 tabset tabset-fade">
<h2 class="tabset tabset-fade">Heatmap of DMRs overlap with
promoters</h2>
<pre class="r"><code>annots = c(&#39;hg19_genes_promoters&#39;)
annots_gr_orig = build_annotations(genome = &#39;hg19&#39;, annotations = annots)

for (comp in names(dmrs)) {

  regions &lt;- makeGRangesFromDataFrame(dmrs[[comp]], keep.extra.columns = T)
  regions &lt;- subsetByOverlaps(regions, annots_gr_orig)
  
  #regions[order(abs(regions$diff.Methy), decreasing = T)[1:min(length(regions), 20000)],]
 mat1 &lt;- as.data.frame(methrix::get_region_summary(meth, regions[order(abs(regions$diff.Methy), decreasing = T)[1:min(length(regions), 20000)],]))
 
  
  plot_mat &lt;- mat1[complete.cases(mat1),-(1:5)]
  
  if (nrow(plot_mat)&gt;1){
 
  cat(&quot;\n&quot;)
  cat(&quot;### Comparisons&quot;,
      comp,
      &quot;\n&quot;)
   pheatmap::pheatmap(plot_mat, annotation_col = as.data.frame(meth@colData)[,c(&quot;Sample.Type&quot;, &quot;Cell.Type&quot;)], colorRampPalette(Cellpaper, space = &quot;rgb&quot;)(10), main = comp, show_rownames = FALSE)
   cat(&quot;\n&quot;)

} else {
  cat(&quot;No overlapping regions with promoters \n&quot;)
}

}</code></pre>
<div id="comparisons-tumor-vs.-normal-in-cmp-cells-1"
class="section level3">
<h3>Comparisons tumor vs. normal in CMP cells</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps_2-1.png" width="672" style="display: block; margin: auto;" />
No overlapping regions with promoters</p>
</div>
<div id="comparisons-cmp-vs.-all-in-normal-1" class="section level3">
<h3>Comparisons CMP vs. all in normal</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps_2-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-tumor-vs.-normal-in-hsc-cells-1"
class="section level3">
<h3>Comparisons tumor vs. normal in HSC cells</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps_2-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-hsc-vs.-all-in-tumor-1" class="section level3">
<h3>Comparisons HSC vs. all in tumor</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps_2-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-hsc-vs.-all-in-normal-1" class="section level3">
<h3>Comparisons HSC vs. all in normal</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps_2-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-tumor-vs.-normal-in-mpp-cells-1"
class="section level3">
<h3>Comparisons tumor vs. normal in MPP cells</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps_2-6.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-mpp-vs.-all-in-tumor-1" class="section level3">
<h3>Comparisons MPP vs. all in tumor</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps_2-7.png" width="672" style="display: block; margin: auto;" />
No overlapping regions with promoters</p>
</div>
<div id="comparisons-tumor-vs.-all-in-normal-1" class="section level3">
<h3>Comparisons tumor vs. all in normal</h3>
<p><img src="figure/DMR_report.Rmd/dmr_heatmaps_2-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="cell-type-specific-dmrs" class="section level2">
<h2>Cell-type specific dmrs</h2>
<div
id="identify-cell-type-specific-dmrs-in-normal-samples-and-plot-all-the-samples."
class="section level3">
<h3>Identify cell-type specific dmrs in normal samples and plot all the
samples.</h3>
<pre class="r"><code>cell_type_specific &lt;- c(&quot;CMP vs. all in normal&quot;, &quot;HSC vs. all in normal&quot;, &quot;MPP vs. all in normal&quot;)

regions_all &lt;- GRanges()

for (comp in cell_type_specific) {
 
  regions &lt;- makeGRangesFromDataFrame(dmrs[[comp]][order(dmrs[[comp]]$diff.Methy, decreasing = T),], keep.extra.columns = T)
  regions$cells &lt;- substr(comp, 1, 3)
  regions_all &lt;- c(regions_all, regions)
  
}

 mat1 &lt;- as.data.frame(get_region_summary(meth, regions_all))
 #mat1$cells &lt;- regions_all$cells
 cells &lt;- regions_all$cells[mat1$rid[which(complete.cases(mat1))]]
  plot_mat &lt;- mat1[complete.cases(mat1),-(1:5)]
  
  

  anno1 &lt;- ggsci::pal_jama(&quot;default&quot;)(4)
  anno2 &lt;- ggsci::pal_jama(&quot;default&quot;)(7)[3:7]
  names(anno1) &lt;- unique(meth@colData$Cell.Type)
  names(anno2) &lt;- unique(as.character(meth@colData$Group))
  ha = HeatmapAnnotation(cell_type = as.data.frame(meth@colData)$Cell.Type,
                         group = as.character(as.data.frame(meth@colData)$Group),
    col = list(cell_type=anno1, group=anno2))
  
  
  Heatmap(plot_mat, name = &quot;Methylation&quot;, row_split = cells, col = colorRampPalette(Cellpaper, space = &quot;rgb&quot;)(10),cluster_rows = TRUE, show_row_names = FALSE, column_title = &quot;Cell type specific signals&quot;,top_annotation = ha, show_column_names = FALSE)</code></pre>
<p><img src="figure/DMR_report.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  cat (&quot;\n&quot;)</code></pre>
</div>
<div
id="identify-cell-type-specific-dmrs-in-tumor-samples-and-plot-all-the-samples."
class="section level3">
<h3>Identify cell-type specific dmrs in tumor samples and plot all the
samples.</h3>
<pre class="r"><code>cell_type_specific &lt;- c(&quot;CMP vs. all in tumor&quot;, &quot;HSC vs. all in tumor&quot;, &quot;MPP vs. all in tumor&quot;)

regions_all &lt;- GRanges()

for (comp in cell_type_specific) {

  regions &lt;- makeGRangesFromDataFrame(dmrs[[comp]][order(dmrs[[comp]]$diff.Methy, decreasing = T),], keep.extra.columns = T)
  regions$cells &lt;- substr(comp, 1, 3)
  regions_all &lt;- c(regions_all, regions)
  
}

 mat1 &lt;- as.data.frame(get_region_summary(meth, regions_all))
 #mat1$cells &lt;- regions_all$cells
 cells &lt;- regions_all$cells[mat1$rid[which(complete.cases(mat1))]]
  plot_mat &lt;- mat1[complete.cases(mat1),-(1:5)]
  
  

  anno1 &lt;- ggsci::pal_jama(&quot;default&quot;)(4)
  anno2 &lt;- ggsci::pal_jama(&quot;default&quot;)(7)[3:7]
  names(anno1) &lt;- unique(meth@colData$Cell.Type)
  names(anno2) &lt;- unique(as.character(meth@colData$Group))
  ha = HeatmapAnnotation(cell_type = as.data.frame(meth@colData)$Cell.Type,
                         group = as.character(as.data.frame(meth@colData)$Group),
    col = list(cell_type=anno1, group=anno2))
  
  
  Heatmap(plot_mat, name = &quot;Methylation&quot;, row_split = cells, col = colorRampPalette(Cellpaper, space = &quot;rgb&quot;)(10),cluster_rows = TRUE, show_row_names = FALSE, column_title = &quot;Cell type specific signals&quot;,top_annotation = ha, show_column_names = FALSE)</code></pre>
<p><img src="figure/DMR_report.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  cat (&quot;\n&quot;)</code></pre>
</div>
</div>
</div>
<div id="relation-of-dmrs-with-functional-regions"
class="section level1">
<h1>Relation of DMRs with functional regions</h1>
<div id="location-of-dmrs-in-relation-to-genes" class="section level2">
<h2>Location of DMRs in relation to genes</h2>
<pre class="r"><code>## loading packages
txdb &lt;- TxDb.Hsapiens.UCSC.hg19.knownGene


dmrs_split &lt;- list()
for (comp in names(dmrs)){
  
  dmrs_split[[paste0(comp, &quot;_up&quot;)]] &lt;- makeGRangesFromDataFrame(dmrs[[comp]][dmrs[[comp]]$diff.Methy&lt;0,], keep.extra.columns = T)
  dmrs_split[[paste0(comp, &quot;_down&quot;)]] &lt;- makeGRangesFromDataFrame(dmrs[[comp]][dmrs[[comp]]$diff.Methy&gt;0,], keep.extra.columns = T)
  
}

peakAnnoList &lt;- lapply(dmrs_split, annotatePeak, TxDb=txdb,
                       tssRegion=c(-3000, 3000), verbose=FALSE)
for (i in 1:length(peakAnnoList)) {
  write.table(as.data.frame(peakAnnoList[[i]]@anno), file=paste0(&quot;data/&quot;, paste0(gsub(&quot; &quot;, &quot;_&quot;, names(peakAnnoList)[[i]])), &quot;.txt&quot;), sep=&quot;\t&quot;, row.names=F, quote = F)

}

plotAnnoBar(peakAnnoList)</code></pre>
<p><img src="figure/DMR_report.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##Distance from TSS

plotDistToTSS(peakAnnoList,
              title=&quot;Distribution of DMRs\nrelative to TSS&quot;)</code></pre>
<p><img src="figure/DMR_report.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="location-of-dmrs-in-relation-to-functional-regions-in-h1hesc-cells"
class="section level2">
<h2>Location of DMRS in relation to functional regions in H1hesc
cells</h2>
<p>Direction “up” means that the methylation is higher in the second
group. “Down” means it is lower in the second group.</p>
<pre class="r"><code> annots = c(&#39;hg19_H1hesc-chromatin&#39;)
  annots_gr_orig = build_annotations(genome = &#39;hg19&#39;, annotations = annots)
 
   plots &lt;-  region_annotation_function(dmrs = dmrs, annots_gr = annots_gr_orig, genome = &quot;hg19&quot;)</code></pre>
<pre class="r"><code>   plots[[1]]</code></pre>
<p><img src="figure/DMR_report.Rmd/unnamed-chunk-6-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>   plots[[2]]</code></pre>
<p><img src="figure/DMR_report.Rmd/unnamed-chunk-6-2.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div
id="location-of-dmrs-in-relation-to-functional-regions-in-k562-cells"
class="section level2">
<h2>Location of DMRS in relation to functional regions in K562
cells</h2>
<p>Direction “up” means that the methylation is higher in the second
group. “Down” means it is lower in the second group.</p>
<pre class="r"><code> annots = c(&#39;hg19_K562-chromatin&#39;)
  annots_gr_orig = build_annotations(genome = &#39;hg19&#39;, annotations = annots)
 
   plots &lt;-  region_annotation_function(dmrs = dmrs, annots_gr = annots_gr_orig, genome = &quot;hg19&quot;)</code></pre>
<pre class="r"><code>   plots[[1]]</code></pre>
<p><img src="figure/DMR_report.Rmd/unnamed-chunk-7-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>   plots[[2]]</code></pre>
<p><img src="figure/DMR_report.Rmd/unnamed-chunk-7-2.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div
id="kegg-pathway-enrichment-of-the-differentially-methylated-regions"
class="section level2 tabset tabset-fade">
<h2 class="tabset tabset-fade">KEGG Pathway enrichment of the
differentially methylated regions</h2>
<p>Direction “up” means that the methylation is higher in the second
group. “Down” means it is lower in the second group.</p>
<pre class="r"><code>library(ChIPseeker)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
txdb &lt;- TxDb.Hsapiens.UCSC.hg19.knownGene
library(ReactomePA)</code></pre>
<pre><code>ReactomePA v1.42.0  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/

If you use ReactomePA in published research, please cite:
Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization. Molecular BioSystems 2016, 12(2):477-479</code></pre>
<pre class="r"><code>library(clusterProfiler)

for (comp in names(dmrs)){
  dmr &lt;- makeGRangesFromDataFrame(dmrs[[comp]], keep.extra.columns = T)

genes &lt;- lapply(list(&quot;Up&quot;=dmr[dmr$diff.Methy&lt;0,], &quot;Down&quot; =dmr[dmr$diff.Methy&gt;0,]), seq2gene_func, tssRegion = c(-3000, 3000), flankDistance = 3000, TxDb=txdb)

#names(genes) &lt;- gsub(&quot; &quot;, &quot;\n&quot;, names(genes))
comppthw &lt;- compareCluster(geneCluster   = genes,
                         fun           = &quot;enrichPathway&quot;,
                         pvalueCutoff  = 0.05,
                         pAdjustMethod = &quot;BH&quot;)

  cat(&quot;\n&quot;)
  cat(&quot;### Comparisons&quot;,
      comp,
      &quot;\n&quot;)
    if (!is.null(comppthw))
print(dotplot(comppthw))
  cat(&quot;\n&quot;)
}</code></pre>
<div id="comparisons-tumor-vs.-normal-in-cmp-cells-2"
class="section level3">
<h3>Comparisons tumor vs. normal in CMP cells</h3>
<p><img src="figure/DMR_report.Rmd/pathway_enrichment-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-cmp-vs.-all-in-tumor-1" class="section level3">
<h3>Comparisons CMP vs. all in tumor</h3>
</div>
<div id="comparisons-cmp-vs.-all-in-normal-2" class="section level3">
<h3>Comparisons CMP vs. all in normal</h3>
<p><img src="figure/DMR_report.Rmd/pathway_enrichment-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-tumor-vs.-normal-in-hsc-cells-2"
class="section level3">
<h3>Comparisons tumor vs. normal in HSC cells</h3>
<p><img src="figure/DMR_report.Rmd/pathway_enrichment-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-hsc-vs.-all-in-tumor-2" class="section level3">
<h3>Comparisons HSC vs. all in tumor</h3>
</div>
<div id="comparisons-hsc-vs.-all-in-normal-2" class="section level3">
<h3>Comparisons HSC vs. all in normal</h3>
<p><img src="figure/DMR_report.Rmd/pathway_enrichment-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-tumor-vs.-normal-in-mpp-cells-2"
class="section level3">
<h3>Comparisons tumor vs. normal in MPP cells</h3>
<p><img src="figure/DMR_report.Rmd/pathway_enrichment-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-mpp-vs.-all-in-tumor-2" class="section level3">
<h3>Comparisons MPP vs. all in tumor</h3>
<p><img src="figure/DMR_report.Rmd/pathway_enrichment-6.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-mpp-vs.-all-in-normal-1" class="section level3">
<h3>Comparisons MPP vs. all in normal</h3>
<p><img src="figure/DMR_report.Rmd/pathway_enrichment-7.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-tumor-vs.-all-in-normal-2" class="section level3">
<h3>Comparisons tumor vs. all in normal</h3>
<p><img src="figure/DMR_report.Rmd/pathway_enrichment-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="go-pathway-enrichment-of-the-differentially-methylated-regions"
class="section level2 tabset tabset-fade">
<h2 class="tabset tabset-fade">GO Pathway enrichment of the
differentially methylated regions</h2>
<p>Direction “up” means that the methylation is higher in the second
group. “Down” means it is lower in the second group.</p>
<pre class="r"><code>library(ChIPseeker)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
txdb &lt;- TxDb.Hsapiens.UCSC.hg19.knownGene
library(ReactomePA)
library(clusterProfiler)

for (comp in names(dmrs)){
  dmr &lt;- makeGRangesFromDataFrame(dmrs[[comp]], keep.extra.columns = T)

genes &lt;- lapply(list(&quot;Up&quot;=dmr[dmr$diff.Methy&lt;0,], &quot;Down&quot; =dmr[dmr$diff.Methy&gt;0,]), seq2gene_func, tssRegion = c(-3000, 3000), flankDistance = 3000, TxDb=txdb)

#names(genes) &lt;- gsub(&quot; &quot;, &quot;\n&quot;, names(genes))
comppthw &lt;- compareCluster(geneCluster   = genes,
                         fun           = &quot;enrichGO&quot;,
                         pvalueCutoff  = 0.05,
                         pAdjustMethod = &quot;BH&quot;, 
                         OrgDb=&#39;org.Hs.eg.db&#39;)

  cat(&quot;\n&quot;)
  cat(&quot;### Comparisons&quot;,
      comp,
      &quot;\n&quot;)
  if (!is.null(comppthw))
print(dotplot(comppthw))
  cat(&quot;\n&quot;)
}</code></pre>
<div id="comparisons-tumor-vs.-normal-in-cmp-cells-3"
class="section level3">
<h3>Comparisons tumor vs. normal in CMP cells</h3>
<p><img src="figure/DMR_report.Rmd/pathway_enrichment_GO-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-cmp-vs.-all-in-tumor-2" class="section level3">
<h3>Comparisons CMP vs. all in tumor</h3>
</div>
<div id="comparisons-cmp-vs.-all-in-normal-3" class="section level3">
<h3>Comparisons CMP vs. all in normal</h3>
<p><img src="figure/DMR_report.Rmd/pathway_enrichment_GO-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-tumor-vs.-normal-in-hsc-cells-3"
class="section level3">
<h3>Comparisons tumor vs. normal in HSC cells</h3>
<p><img src="figure/DMR_report.Rmd/pathway_enrichment_GO-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-hsc-vs.-all-in-tumor-3" class="section level3">
<h3>Comparisons HSC vs. all in tumor</h3>
<p><img src="figure/DMR_report.Rmd/pathway_enrichment_GO-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-hsc-vs.-all-in-normal-3" class="section level3">
<h3>Comparisons HSC vs. all in normal</h3>
<p><img src="figure/DMR_report.Rmd/pathway_enrichment_GO-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-tumor-vs.-normal-in-mpp-cells-3"
class="section level3">
<h3>Comparisons tumor vs. normal in MPP cells</h3>
<p><img src="figure/DMR_report.Rmd/pathway_enrichment_GO-6.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-mpp-vs.-all-in-tumor-3" class="section level3">
<h3>Comparisons MPP vs. all in tumor</h3>
<p><img src="figure/DMR_report.Rmd/pathway_enrichment_GO-7.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-mpp-vs.-all-in-normal-2" class="section level3">
<h3>Comparisons MPP vs. all in normal</h3>
<p><img src="figure/DMR_report.Rmd/pathway_enrichment_GO-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparisons-tumor-vs.-all-in-normal-3" class="section level3">
<h3>Comparisons tumor vs. all in normal</h3>
<p><img src="figure/DMR_report.Rmd/pathway_enrichment_GO-9.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] ReactomePA_1.42.0                      
 [2] org.Hs.eg.db_3.16.0                    
 [3] clusterProfiler_4.6.0                  
 [4] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
 [5] GenomicFeatures_1.49.7                 
 [6] AnnotationDbi_1.59.1                   
 [7] ChIPseeker_1.34.0                      
 [8] ComplexHeatmap_2.14.0                  
 [9] annotatr_1.23.0                        
[10] BSgenome.Hsapiens.UCSC.hg19_1.4.3      
[11] BSgenome_1.66.0                        
[12] rtracklayer_1.57.0                     
[13] Biostrings_2.65.6                      
[14] XVector_0.37.1                         
[15] BiocManager_1.30.19                    
[16] scales_1.2.1                           
[17] plotly_4.10.0                          
[18] ggplot2_3.3.6                          
[19] DSS_2.46.0                             
[20] bsseq_1.34.0                           
[21] BiocParallel_1.31.12                   
[22] pheatmap_1.0.12                        
[23] methrix_1.12.0                         
[24] SummarizedExperiment_1.27.3            
[25] Biobase_2.57.1                         
[26] GenomicRanges_1.49.0                   
[27] GenomeInfoDb_1.33.7                    
[28] IRanges_2.31.2                         
[29] S4Vectors_0.35.4                       
[30] BiocGenerics_0.43.4                    
[31] MatrixGenerics_1.9.1                   
[32] matrixStats_0.62.0                     
[33] data.table_1.14.4                      
[34] workflowr_1.7.0                        

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3                R.methodsS3_1.8.2            
  [3] tidyr_1.2.1                   bit64_4.0.5                  
  [5] knitr_1.40                    DelayedArray_0.23.2          
  [7] R.utils_2.12.1                KEGGREST_1.38.0              
  [9] RCurl_1.98-1.9                doParallel_1.0.17            
 [11] generics_0.1.3                callr_3.7.3                  
 [13] cowplot_1.1.1                 RSQLite_2.2.18               
 [15] shadowtext_0.1.2              bit_4.0.4                    
 [17] tzdb_0.3.0                    enrichplot_1.18.0            
 [19] xml2_1.3.3                    httpuv_1.6.6                 
 [21] ggsci_2.9                     assertthat_0.2.1             
 [23] viridis_0.6.2                 xfun_0.34                    
 [25] hms_1.1.2                     jquerylib_0.1.4              
 [27] evaluate_0.17                 promises_1.2.0.1             
 [29] fansi_1.0.3                   restfulr_0.0.15              
 [31] progress_1.2.2                caTools_1.18.2               
 [33] dbplyr_2.2.1                  igraph_1.3.5                 
 [35] DBI_1.1.3                     htmlwidgets_1.5.4            
 [37] purrr_0.3.4                   ellipsis_0.3.2               
 [39] crosstalk_1.2.0               dplyr_1.0.10                 
 [41] permute_0.9-7                 biomaRt_2.54.0               
 [43] sparseMatrixStats_1.9.0       vctrs_0.5.0                  
 [45] Cairo_1.6-0                   cachem_1.0.6                 
 [47] withr_2.5.0                   ggforce_0.4.1                
 [49] HDO.db_0.99.1                 vroom_1.6.0                  
 [51] GenomicAlignments_1.33.1      treeio_1.22.0                
 [53] prettyunits_1.1.1             cluster_2.1.4                
 [55] DOSE_3.24.0                   ape_5.6-2                    
 [57] lazyeval_0.2.2                crayon_1.5.2                 
 [59] pkgconfig_2.0.3               labeling_0.4.2               
 [61] tweenr_2.0.2                  nlme_3.1-160                 
 [63] rlang_1.0.6                   lifecycle_1.0.3              
 [65] downloader_0.4                filelock_1.0.2               
 [67] BiocFileCache_2.6.0           AnnotationHub_3.6.0          
 [69] rprojroot_2.0.3               polyclip_1.10-4              
 [71] graph_1.76.0                  Matrix_1.5-1                 
 [73] aplot_0.1.8                   Rhdf5lib_1.19.2              
 [75] boot_1.3-28                   whisker_0.4                  
 [77] GlobalOptions_0.1.2           processx_3.7.0               
 [79] png_0.1-7                     viridisLite_0.4.1            
 [81] rjson_0.2.21                  bitops_1.0-7                 
 [83] getPass_0.2-2                 R.oo_1.25.0                  
 [85] gson_0.0.9                    KernSmooth_2.23-20           
 [87] rhdf5filters_1.9.0            blob_1.2.3                   
 [89] DelayedMatrixStats_1.20.0     shape_1.4.6                  
 [91] stringr_1.4.1                 qvalue_2.30.0                
 [93] regioneR_1.30.0               readr_2.1.3                  
 [95] gridGraphics_0.5-1            reactome.db_1.82.0           
 [97] graphite_1.44.0               memoise_2.0.1                
 [99] magrittr_2.0.3                plyr_1.8.7                   
[101] gplots_3.1.3                  zlibbioc_1.43.0              
[103] compiler_4.2.2                scatterpie_0.1.8             
[105] BiocIO_1.8.0                  RColorBrewer_1.1-3           
[107] plotrix_3.8-2                 clue_0.3-62                  
[109] Rsamtools_2.13.4              cli_3.3.0                    
[111] patchwork_1.1.2               ps_1.7.0                     
[113] MASS_7.3-58.1                 tidyselect_1.2.0             
[115] stringi_1.7.8                 highr_0.9                    
[117] yaml_2.3.6                    GOSemSim_2.24.0              
[119] locfit_1.5-9.6                ggrepel_0.9.1                
[121] sass_0.4.2                    fastmatch_1.1-3              
[123] tools_4.2.2                   circlize_0.4.15              
[125] rstudioapi_0.14               foreach_1.5.2                
[127] git2r_0.30.1                  gridExtra_2.3                
[129] farver_2.1.1                  ggraph_2.1.0                 
[131] digest_0.6.29                 shiny_1.7.3                  
[133] Rcpp_1.0.9                    BiocVersion_3.16.0           
[135] later_1.3.0                   httr_1.4.4                   
[137] colorspace_2.0-3              XML_3.99-0.12                
[139] fs_1.5.2                      splines_4.2.2                
[141] yulab.utils_0.0.5             tidytree_0.4.1               
[143] graphlayouts_0.8.3            ggplotify_0.1.0              
[145] xtable_1.8-4                  jsonlite_1.8.3               
[147] ggtree_3.6.0                  tidygraph_1.2.2              
[149] UpSetR_1.4.0                  ggfun_0.0.7                  
[151] R6_2.5.1                      pillar_1.8.1                 
[153] htmltools_0.5.3               mime_0.12                    
[155] glue_1.6.2                    fastmap_1.1.0                
[157] interactiveDisplayBase_1.36.0 codetools_0.2-18             
[159] fgsea_1.24.0                  utf8_1.2.2                   
[161] lattice_0.20-45               bslib_0.4.1                  
[163] tibble_3.1.8                  curl_4.3.3                   
[165] gtools_3.9.3                  magick_2.7.3                 
[167] GO.db_3.16.0                  limma_3.53.10                
[169] rmarkdown_2.17                munsell_0.5.0                
[171] GetoptLong_1.0.5              rhdf5_2.41.1                 
[173] GenomeInfoDbData_1.2.9        iterators_1.0.14             
[175] HDF5Array_1.25.2              reshape2_1.4.4               
[177] gtable_0.3.1                 </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
