---
title: "read_in"
author: "Reka_Toth"
date: '`r format(Sys.time(), "%d %B, %Y")`'
output:
 # html_notebook: default
  workflowr::wflow_html:
    toc: false
    code_folding: "hide"
#output: workflowr::wflow_html:
editor_options:
  chunk_output_type: console
---

## Analysis of Gene expression in methylation subgroups

Based on the methylation analysis, two patients, 6515 and 8561 were identified as having a lower similarity to the erythroid progentior cells than the rest of the samples. There is also a difference between their gene expression pattern. Please keep in mind, that the gender composition of these two groups are imbalanced, so probably some of the genes are gender related. Most of the plots and analysis only contains genes with known chromosomal location on the autosomes. 


```{r include=F}
knitr::opts_chunk$set(message = F, warning = F)
```



```{r message=FALSE, warning=FALSE}

##libraries


library(DESeq2)
library("GeneTonic")
library("org.Hs.eg.db")
library("AnnotationDbi")
library(pcaExplorer)
library(topGO)
library('TxDb.Hsapiens.UCSC.hg19.knownGene')
library(Glimma)
library(dplyr)
library(GeneTonic)
library(DT)
library(plotly)
library(scales)
library(clusterProfiler)
library(pheatmap)

```

```{r}
#get_aggrscores
get_aggrscores_mod <-  function (res_enrich, res_de, annotation_obj, gtl = NULL, aggrfun = mean) 
{
    if (!is.null(gtl)) {
        checkup_gtl(gtl)
        dds <- gtl$dds
        res_de <- gtl$res_de
        res_enrich <- gtl$res_enrich
        annotation_obj <- gtl$annotation_obj
    }
    gs_expanded <- tidyr::separate_rows(res_enrich, "gs_genes", 
        sep = ",")
    gs_expanded$log2FoldChange <- res_de[annotation_obj$ENSEMBL[match(gs_expanded$gs_genes, 
        annotation_obj$gene_name)], ]$log2FoldChange
    gs_aggregated <- lapply(seq_len(nrow(res_enrich)), function(i) {
        this_gsid <- res_enrich$gs_id[i]
        this_subset <- gs_expanded[gs_expanded$gs_id == this_gsid, 
            ]
        upgenes <- sum(this_subset$log2FoldChange > 0)
        downgenes <- sum(this_subset$log2FoldChange < 0)
        z_score <- (upgenes - downgenes)/sqrt(upgenes + downgenes)
        aggr_score <- aggrfun(this_subset$log2FoldChange)
        return(c(DE_count = nrow(this_subset), Z_score = z_score, 
            aggr_score = aggr_score))
    })
    names(gs_aggregated) <- res_enrich$gs_id
    res_enrich$DE_count <- vapply(gs_aggregated, "[", 1, FUN.VALUE = numeric(1))
    res_enrich$z_score <- vapply(gs_aggregated, "[", 2, FUN.VALUE = numeric(1))
    res_enrich$aggr_score <- vapply(gs_aggregated, "[", 3, FUN.VALUE = numeric(1))
    return(res_enrich)
}


gs_summary_heat_mod <- 
function (res_enrich, res_de, annotation_obj, gtl = NULL, n_gs = 80) {
    if (!is.null(gtl)) {
        checkup_gtl(gtl)
        dds <- gtl$dds
        res_de <- gtl$res_de
        res_enrich <- gtl$res_enrich
        annotation_obj <- gtl$annotation_obj
    }
    res_enrich2 <- res_enrich[seq_len(n_gs), ]
    gs_expanded <- tidyr::separate_rows(res_enrich2, "gs_genes", 
        sep = ",")
    gs_expanded$log2FoldChange <- res_de[annotation_obj$ENSEMBL[match(gs_expanded$gs_genes, 
        annotation_obj$gene_name)], ]$log2FoldChange
    gs_expanded[["gs_id"]] <- factor(gs_expanded[["gs_id"]], 
        levels = res_enrich2[["gs_id"]])
    gs_expanded[["gs_description"]] <- factor(gs_expanded[["gs_description"]], 
        levels = res_enrich2[["gs_description"]])
    gs_expanded[["gs_genes"]] <- factor(gs_expanded[["gs_genes"]], 
        levels = unique(gs_expanded[["gs_genes"]]))
    p <- ggplot(gs_expanded, aes_string(x = "gs_genes", y = "gs_description")) + 
        geom_tile(aes_string(fill = "log2FoldChange"), color = "white") + 
        scale_fill_gradient2(low = muted("deepskyblue"), mid = "lightyellow", 
            high = muted("firebrick"), name = "log2 Fold Change") + 
        xlab(NULL) + ylab(NULL) + theme_minimal() + theme(panel.grid.major = element_blank(), 
        axis.text.x = element_text(angle = 75, hjust = 1))
    return(p)
}


```


```{r}

####################with count data, using DESeq2

####the count data is produced by stringtie, using prepDE.py
##transcript level analysis is also possible


gene_count_matrix <- read.csv( "data/gene_count_matrix.csv", row.names=1)


    
 anno <- data.frame(Patient.ID= gsub("X([[:digit:]]{4}).BM[[:punct:]]?(CMP|MPP|sorted.Bilk|latest)?_S[[:digit:]]+", "\\1",  colnames(gene_count_matrix)), 
                    Group=gsub("X([[:digit:]]{4}).BM[[:punct:]]?(CMP|MPP|sorted.Bilk|latest)?_S[[:digit:]]+", "\\2",  colnames(gene_count_matrix)) %>% ifelse(. %in% c("CMP", "MPP"), ., "Bulk"), Cell.Type="BM", Age=NA)
 
rownames(anno) <-  colnames(gene_count_matrix)
anno$meth_group <- NA
anno$meth_group[grep("8561|6515",rownames(anno))] <- "group1"
anno$meth_group[-grep("8561|6515",rownames(anno))] <- "group2"


rownames(gene_count_matrix) <- gsub("(ENSG[[:digit:]]*\\.[[:digit:]]*)(\\|.*)", "\\1", rownames(gene_count_matrix))
  
dds <- DESeqDataSetFromMatrix(gene_count_matrix, colData = anno, design = ~ meth_group, )
#dds <- DESeq(dds)
#saveRDS(dds, "data/deseq.RDS")
#glimmaMDS(dds, html = "output/MDS.html")

#plotPCA(vsd, intgroup=c("PatientID", "SampleType"))

glimmaMDS(dds, html = "output/MDS.html")

rownames(dds) <- substr(rownames(dds),1,15)
dds <- dds[!duplicated(rownames(dds)),]


```



```{r}


dds<- DESeq(dds)
res <- results(dds)

anno_genes <- data.frame(
  ensembl = rownames(dds),
  gene_id =   mapIds(org.Hs.eg.db,
                            keys= rownames(dds),
                            column="ENTREZID",
                            keytype="ENSEMBL",
                            multiVals="first"),
  gene_name =  mapIds(org.Hs.eg.db,
                            keys= rownames(dds),
                            column="SYMBOL",
                            keytype="ENSEMBL",
                            multiVals="first"),

  stringsAsFactors = FALSE,
  row.names =  rownames(dds)
)


anno_genes$gene_chr <- mapIds(TxDb.Hsapiens.UCSC.hg19.knownGene,
                            keys= anno_genes$gene_id,
                            column="TXCHROM",
                            keytype="GENEID",
                            multiVals="first")

  colnames(anno_genes) <- c("ENSEMBL", "GeneID", "GeneName", "GeneCHR")
  

```

```{r, fig.height=8, fig.width=10, results='asis'}

gene_names <- c("CD34", "THY1", "PROM1", "EPO", "KITLG", "FLT3LG", "CSF2", "CSF1", "IL3", "IL6", "TPO", "MEIS1", "MEIS2")

vst <- getVarianceStabilizedData(dds)

vst_cand <- vst[anno_genes$ENSEMBL[which(anno_genes$GeneName %in% gene_names)],]
rownames(vst_cand) <- anno_genes$GeneName[which(anno_genes$GeneName %in% gene_names)]

vst_hox <- vst[anno_genes$ENSEMBL[grep("HOX|CDX", anno_genes$GeneName)],]
rownames(vst_hox) <- anno_genes$GeneName[grep("HOX|CDX", anno_genes$GeneName)]

vst_cand <- vst_cand[apply(vst_cand, 1, sd, na.rm=T)!=0,]
pheatmap(vst_cand, scale = "row", main = "Candidate genes")

vst_hox <- vst_hox[apply(vst_hox, 1, sd, na.rm=T)!=0,]
pheatmap(vst_hox, scale = "row", main = "HOX and CDXgenes")


res <- results(dds)

  glMDPlot(res, counts=assay(dds), anno = anno_genes, groups = dds$meth_group, main = "Meth_groups", status = as.numeric(res$padj<0.05), launch = F,  transform = F, html = "meth_groups",  folder = "/glimma-plots")

  res %>%
    as.data.frame() %>%
    mutate(across(where(is.numeric), ~round(.x,digits=2))) %>%
DT::datatable(., caption = "Gene expression results", filter = 'top', options = list(scrollX = '400px'))


  
```


```{r, results='asis'}

bg_ids <- rownames(dds)[rowSums(counts(dds)) > 0]
bg_symbols <- anno_genes[bg_ids, "GeneName"]


sig_res <- as.data.frame(res[which(res$padj<0.05),])
sig_res <- merge(sig_res, anno_genes, by.x="row.names", by.y="ENSEMBL", all.x=T)
sig_res <- sig_res[-which(sig_res$GeneCHR %in% c("chrX", "chrY")),]

enr <- topGOtable(sig_res$GeneName, bg_symbols,
                             ontology = "BP",
                             mapping = "org.Hs.eg.db",
                             geneID = "symbol")

  colnames(anno_genes) <- c("ENSEMBL", "gene_id", "gene_name", "GeneCHR")


res_enrich <- shake_topGOtableResult(enr)
res_enrich_aggr <- get_aggrscores_mod(res_enrich, res, anno_genes)
res_enrich_simplified <- gs_simplify(res_enrich_aggr)
res_enrich_simplified$gs_description <- substr(res_enrich_simplified$gs_description, 1, 25)

res_DT <- as.data.frame(res_enrich_simplified)
cat("\n")  
cat("### Methylation groups, enrichment analysis \n")
    
cat("\n")  

DT::datatable(res_enrich_simplified, caption = "Methylation groups", options = list(scrollX = '400px'))




```

```{r}

gs <- gs_summary_overview(res_enrich_simplified)
print(gs)

#gs_summary_heat_mod(res_enrich_aggr,
#res,
#anno_genes,
#n_gs = 15)

gs_dendro(res_enrich_aggr,
n_gs = 50,
gs_dist_type = "kappa",
clust_method = "ward.D2",
color_leaves_by = "z_score",
size_leaves_by = "gs_pvalue",
color_branches_by = "clusters",
create_plot = TRUE)

gs_alluvial(res_enrich = res_enrich,
res_de = res,
annotation_obj = anno_genes,
n_gs = 4)


```

